R/autosyn.R

```diff
@@ -472,3 +472,210 @@ flywire_ntplot3d <- function(x, nts=c("gaba", "acetylcholine", "glutamate",
+#'   has a \code{data.frame} of synapses at neuron$connectors.
+#'
+#' @export
+#' @family automatic-synapses
+#'
+#' @examples
+#' \donttest{
+#' \dontrun{
+#' choose_segmentation("flywire")
+#' nx=xform_brain(elmr::dense_core_neurons, ref="FlyWire", sample="FAFB14")
+#' xyz = xyzmatrix(nx)
+#' ids = unique(flywire_xyz2id(xyz[sample(nrow(xyz),100),]))
+#' neurons = skeletor(ids, brain = elmr::FAFB14.surf)
+#' neurons.syns = flywire_neurons_add_synapses(neurons, transmitters = TRUE)
+#' neurons.syns[,]
+#'
+#' # Plot in 3D
+#' library(catmaid)
+#' nopen3d()
+#' plot3d(neurons.syns, WithConnectors = TRUE)
+#'
+#' # Axon-dendrite split
+#' library(hemibrainr)
+#' neurons.flow = flow_centrality(neurons.syns,
+#'   polypre = TRUE,
+#'   mode = "centrifugal")
+#' clear3d()
+#' plot3d_split(neurons.flow, WithConnectors = TRUE,
+#' transmitter = TRUE,
+#' radius = 1000, soma = 4000)
+#' }
+#' }
+flywire_neurons_add_synapses <- function(x,
+                                         connectors = NULL,
+                                         cloudvolume.url=NULL, # fafbseg.cloudvolume.url="graphene://https://prodv1.flywire-daf.com/segmentation/1.0/fly_v31"
+                                         method=c("auto", "spine", "sqlite"),
+                                         remove_autapses = TRUE,
+                                         cleft.threshold = 0,
+                                         Verbose=TRUE,
+                                         transmitters=TRUE,
+                                         local = NULL, # "/Volumes/nnautilus/projects/JanFunke"
+                                         ...) UseMethod("flywire_neurons_add_synapses")
+
+#' @export
+#' @rdname flywire_neurons_add_synapses
+flywire_neurons_add_synapses.neuron <- function(x,
+                                                connectors = NULL,
+                                                cloudvolume.url=NULL,
+                                                method=c("auto", "spine", "sqlite"),
+                                                remove_autapses = TRUE,
+                                                cleft.threshold = 0,
+                                                Verbose=TRUE,
+                                                transmitters=TRUE,
+                                                local = NULL,
+                                                ...){
+  method = match.arg(method)
+  rootid = x$flywire.id
+  if(is.null(connectors)){
+    synapses = flywire_partners(rootid = rootid,
+                                partners = "both",
+                                roots=TRUE,
+                                details=TRUE,
+                                cloudvolume.url = cloudvolume.url,
+                                method=method,
+                                Verbose=Verbose,
+                                local = local,
+                                ...)
+    if(isTRUE(!is.null(synapses) && !nrow(synapses))){
```

Comment:
@alexanderbates what are you trying to test for here? What happens if `synapses=NULL`. I think what you want is:

```
if(is.null(synapses) || nrow(synapses)==0) {
...
}
```
You could probably wrap the second test in `isTRUE` to be on the safe side, but I don't there are any other values that need to be considered.

Reply:
changed as suggested